contracts/igp/interchain-gas-paymaster/src/main.sw

```diff
@@ -0,0 +1,212 @@
+contract;
+
+dep interface;
+
+use std::{
+    call_frames::msg_asset_id,
+    constants::BASE_ASSET_ID,
+    context::{
+        msg_amount,
+        this_balance,
+    },
+    logging::log,
+    token::transfer,
+    u128::U128,
+};
+
+use std_lib_extended::{option::*, result::*};
+
+use hyperlane_interfaces::igp::{GasOracle, InterchainGasPaymaster, RemoteGasData};
+
+use ownership::{interface::Ownable, log_ownership_transferred, require_msg_sender};
+
+use interface::{
+    BeneficiarySetEvent,
+    Claimable,
+    ClaimEvent,
+    GasOracleSetEvent,
+    GasPaymentEvent,
+    OnChainFeeQuoting,
+};
+
+/// The scale of a token exchange rate. 1e19.
+const TOKEN_EXCHANGE_RATE_SCALE: u64 = 10000000000000000000;
+
+// TODO: set this at compile / deploy time.
+// NOTE for now this is temporarily set to the address of a PUBLICLY KNOWN
+// PRIVATE KEY, which is the first default account when running fuel-client locally.
+const INITIAL_OWNER: Option<Identity> = Option::Some(Identity::Address(Address::from(0x6b63804cfbf9856e68e5b6e7aef238dc8311ec55bec04df774003a2c96e0418e)));
```

Comment:
btw this heinous stuff will be addressed by https://github.com/hyperlane-xyz/fuel-contracts/issues/54